Define delimiter-encapsulation (#120)
citation-style-language/documentation: specification.rst

      </citation>
    </style>

> The output of a ``<text macro="...">`` does not have delimiters from any

Reviewer: Thanks for working on this! I think we're almost there. I think we can improve on interspersed. 
Does "The output of a ``<text macro="...">`` does not take delimiters from any
ancestor delimiting element" (and similarly below) capture it? 
"Interspersed" sounds like a random process to my (admittedly non-native speaker) ears

Reviewer 2: I'm lost at this point. Can you guys "resolve" the suggestions or comments that you've addressed, once you have, so that we can tell when ready to merge?

Reviewer: From my end this is the only remaining unresolved issue. I've left your proposed change unresolved (since it's yours)

> The output of a ``<text macro="...">`` does not have delimiters from any

Reviewer 3: ```suggestion
Delimiters from any ancestor delimiting element are not applied within the output of <text macro="..."> element (see delimiter).
```

How about that?

Reviewer: I like it (though I think merging the suggestion won't work since it only replaces the first line.

> The output of a ``<text macro="...">`` does not have delimiters from a parent 

Reviewer: should this be `parent` or `ancestor` (honestly not sure)?

> The output of a ``<text macro="...">`` does not have delimiters from any
> ancestor delimiting element interspersed within it (see `delimiter`_).

Reviewer 3: ```suggestion
Delimiters from any ancestor delimiting element are not applied within the output of <text macro="..."> element (see delimiter).
```

-  "none" - element only tests "true" when none of the conditions test "true"
   for any given test value

> The output of ``cs:choose`` (i.e. the output of the matching ``cs:if``,
> ``cs:else-if``, or ``cs:else``) *does* have delimiters from the nearest
> ancestor delimiting element interspersed within it (see `delimiter`_).
> 

Reviewer 3: ```suggestion
Delimiters from the nearest delimiters from the nearest ancestor delimiting element *are* applied within the output of ``cs:choose`` (i.e., the output of the matching ``cs:if``, ``cs:else-if``, or ``cs:else``; see `delimiter`_).

```

Reviewer 3: Using the "delimiters are applied within" phrasing instead of "output has delimiters interspersed" phrasing.

-  "none" - element only tests "true" when none of the conditions test "true"
   for any given test value

The output of ``cs:choose`` (i.e. the output of the matching ``cs:if``, 

> ``cs:else-if``, or ``cs:else``) *does* have delimiters from a parent delimiting 

Reviewer: ditto

Author: In each case ancestor would be technically correct if you did `group with delimiter > choose > if > choose > if` hmm

Author: Nearest ancestor delimiting element does it I think

Author: Because I defined delimiting element to be even if there is no delimiter set on it, yeah

(delimiting names within name lists), ``cs:group`` and ``cs:layout`` (delimiting
the output of the child elements).

> A delimiting element is any element as above which takes a ``delimiter`` 
> attribute, whether the attribute is supplied or not.

Reviewer 3: ```suggestion
A delimiting element is any element as above which takes a ``delimiter`` attribute, whether the attribute is supplied or not.
```

Reviewer 3: Putting single sentences on single lines.

> The output of a delimiting element does not have delimiters from any ancestor
> delimiting element interspersed within it. The following produces ``retrieved:
> <http://example.com>``:

Reviewer 3: ```suggestion
Delimiters from any ancestor delimiting element are not applied within the output of a delimiting element. The following produces ``retrieved: <http://example.com>``:
```

Reviewer 3: Same "interspersed" comment as above.

(delimiting names within name lists), ``cs:group`` and ``cs:layout`` (delimiting
the output of the child elements).

A delimiting element is any element as above which takes a ``delimiter`` 
attribute, whether the attribute is supplied or not.

The output of all delimiting elements is delimiter-encapsulated: no 
``delimiter`` from a parent delimiting element will delimit between multiple 

> outputs of an inner delimiter-encapsulated element. The following produces

Reviewer 3: We've avoided using "shall produce"-type language. When something is required, we either use "must" or just the present tense "produces"

> outputs of an inner delimiter-encapsulated element. The following produces

Reviewer 2: Seems a little weird to me, but OK; as long as consistent.h

> outputs of an inner delimiter-encapsulated element. The following produces

Author: I said inner because it's not necessarily a direct child. `group > choose > if > group` is covered as well. So you could say "outer" instead of "parent".

> ``delimiter`` from a parent delimiting element will delimit between multiple 
> outputs of an inner delimiter-encapsulated element. The following produces

Reviewer 2: ```suggestion
``delimiter`` from an ancestor delimiting element will delimit between multiple 
outputs of a descendant delimiter-encapsulated element. The following produces
```

> ``a; bc``:
> 
> .. sourcecode:: xml
> 
>     <group delimiter="; ">
>       <text value="a" />
>       <group>
>         <text value="b" />
>         <text value="c" />
>       </group>
>     </group>
> 

Reviewer: ```suggestion
``retrieved: <http://example.com``:

.. sourcecode:: xml

    <group delimiter=": ">
      <text term="retrieved" />
      <group>
        <text term="&lt;" />
        <text variable="URL"/>
      </group>
    </group>

```

Reviewer: Happy for other examples, but the specifications use realistic examples throughout and I think we should stick with that both for consistency and for readability

Author: Agree.
